src/RSocketClient.cpp

```diff
@@ -60,6 +63,51 @@ folly::Future<std::unique_ptr<RSocketRequester>> RSocketClient::connect(
   return future;
 }
 
+folly::Future<folly::Unit> RSocketClient::resume() {
+  CHECK(stateMachine_);
+  folly::Promise<folly::Unit> promise;
+  auto future = promise.getFuture();
+  connectionFactory_->connect([ this, promise = std::move(promise) ](
+      std::unique_ptr<DuplexConnection> connection,
+      folly::EventBase & evb) mutable {
+    CHECK(evb_ == &evb);
+
+    class ResumeCallback : public ClientResumeStatusCallback {
+     public:
+      ResumeCallback(folly::Promise<folly::Unit> promise)
+          : promise_(std::move(promise)) {}
+      void onResumeOk() noexcept override {
+        promise_.setValue(folly::Unit());
+      }
+      void onResumeError(folly::exception_wrapper ex) noexcept override {
+        promise_.setException(ex);
+      }
+      void onConnectionError(folly::exception_wrapper ex) noexcept override {
+        promise_.setException(ex);
```

Comment:
Not being able to distinguish a resume error from a connection error will make it impossible to implement retries correctly.